### DISCUSSION TITLE
does pm.batteryLevel only change in intervals of 10%?

### DISCUSSION THREAD

**Discussion by MilesZew** [bitcraze/discussions#516] [Q&A] [answered]
I've recently been messing around with the crazyflie making my own keyboard controller in python, and I wanted to add the ability to check the battery life. However, when I did so, by using the pm.batteryLevel logging variable I only got a value of 10%, until suddenly without warning it dropped to 0 and the crazyflie dropped out of the air.

Of note is a possibly unrelated problem where after charging the crazyflie for an hour it still was only at 10%, and only had about a minute of flight time

**Comment by hmllr** [comment, marked as answer]
Indeed, you only get the pm.batteryLevel in 10% steps (implemented here: https://github.com/bitcraze/crazyflie-firmware/blob/2022.12/src/hal/src/pm_stm32f4.c#L176) However, to prevent the CF to drop out of the air you could use the pmIsBatteryLow() function and land in time (the voltage used as threshold here is a parameter, so configurable - the default is 3.2V). 
Hope this helps,
Hanna
